platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBook.java

```diff
@@ -230,38 +228,67 @@ public long getTotalWeight() {
         return totalWeight;
     }
 
+    /**
+     * Find the NodeId for the member whose address has the given public key. Returns null if it does not exist.
+     *
+     * @param publicKey the public key to look up
+     * @return the NodeId of the member with that key, or null if it was not found
+     */
+    @Nullable
+    public NodeId getNodeId(@NonNull final String publicKey) {
+        Objects.requireNonNull(publicKey, "publicKey must not be null");
+        return publicKeyToId.get(publicKey);
+    }
+
     /**
      * Find the ID for the member whose address has the given public key. for now, this uses the nickname instead of the
      * public key. Returns -1 if it does not exist.
      *
      * @param publicKey the public key to look up
      * @return the ID of the member with that key, or -1 if it was not found
+     * @deprecated use {@link #getNodeId(String)} instead
      */
-    public long getId(final String publicKey) {
-        return publicKeyToId.getOrDefault(publicKey, -1L);
+    @Deprecated(since = "0.39.0", forRemoval = true)
+    public long getId(@NonNull final String publicKey) {
```

Comment:
Is there a reason why we can't remove this method in this PR?

Reply:
nope, removed.